lib/veri_helpers.ml

```diff
@@ -0,0 +1,45 @@
+open Core_kernel.Std
+open Bap.Std
+open Bap_traces.Std
+
+module Dis  = Disasm_expert.Basic
+
+type insn = Dis.full_insn
+
+let lift_insn arch (mem,insn) = 
+  let module Target = (val target_of_arch arch) in
+  match Target.lift mem insn with
+  | Ok bil -> Some bil
+  | Error _ -> None
+
+let insn_of_mem dis mem =
+  let open Or_error in
+  Dis.insn_of_mem dis mem >>= (fun (imem, insn, left) -> 
+      match insn with 
+      | Some insn -> return (imem, insn)
+      | None -> error_string "no instruction in memory")
+
+let insn_of_chunk arch chunk =
```

Comment:
this function is very heavy, you're creating a new instance of disassembler at every frame. This will result in a very slow interpretation. The disassembler should be created only once.